### PR TITLE
[wasm] Build improvements, and fixes

### LINKED ISSUE #60862: WebAssembly <NativeFileReference> isn't included in VS's fast up-to-date check
If you have, for example:

```
<NativeFileReference Include="somefile.a" />
```

... then you can call it via `[DllImport("somefile")]`. That's all great. But if you then do something to modify `somefile.a`, for example rebuild it from sources, then simply running Ctrl+F5 from VS again will **not** be enough to see your changes. That's because VS will consider the .NET project already up-to-date, assuming you didn't edit any .NET files. You have to artificially touch one of your `.cs` files (or similar) to make it rebuild.

### How to fix it

See info at https://github.com/dotnet/project-system/blob/main/docs/up-to-date-check.md#up-to-date-check for how the MSBuild targets should tell VS about the native input/output files and include them in its up-to-date check.

### PR DESCRIPTION
**Pull request by radical**
Includes:
- Better error logging, and handling
- Add @(NativeFileReference) to up-to-date check items for VS
- Add `WasmBuild.sln`
- Better fix up of symbol names for pinvokes, and callbacks, based on @lambdageek's suggestion in https://github.com/dotnet/runtime/pull/60814#discussion_r736903990

Fixes dotnet#60862

### CODE REVIEW

File: src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs [hunk 1]
```diff
@@ -216,6 +231,41 @@ static bool ShouldTreatAsVariadic(PInvoke[] candidates)
...
+        UTF8Encoding utf8 = new();
+        byte[] bytes = utf8.GetBytes(name);
+        StringBuilder sb = new();
+
+        foreach (byte b in bytes)
+        {
+            if ((b >= (byte)'0' && b <= (byte)'9') ||
+                (b >= (byte)'a' && b <= (byte)'z') ||
+                (b >= (byte)'A' && b <= (byte)'Z') ||
+                (b == (byte)'_'))
```
Comment (lambdageek) [inline src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs:245]: So the most common character outside of this range are probably going to be `'.'` and `'+'` for nested `type!.FullName`. I wonder if we should mangle those in some shorter way.

Comment (radical) [reply src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs:245]: ugh.. that's the reason I had retained `    private static char[] s_charsToReplace = new[] { '.', '-', };`, but looks like I messed up the patch when rebasing, merging multiple times!

We could `'+'` to that array, and check that?

Comment (radical) [reply src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs:245]: I mean, if it's in `s_charsToReplace`, then use `_`?

**Review by lambdageek — APPROVED** [review]

**Review by lewing — APPROVED** [review]

**Comment by radical** [comment]
Merged `main` to pick up the fix for https://github.com/dotnet/runtime/issues/62303 .

**Comment by radical** [comment]
Wasm/AOT tests failing due to https://github.com/dotnet/runtime/issues/62362 , added part of https://github.com/dotnet/runtime/pull/62352 .

**Pull request merged**